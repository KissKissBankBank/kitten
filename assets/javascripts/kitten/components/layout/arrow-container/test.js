import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ArrowContainer } from './index'

describe('<ArrowContainer />', () => {
  let component

  describe('with default props', () => {
    beforeEach(() => {
      component = renderer
        .create(<ArrowContainer>Foobar</ArrowContainer>)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `color` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<ArrowContainer color="#ff0000">Foobar</ArrowContainer>)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `size` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<ArrowContainer size={20}>Foobar</ArrowContainer>)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `position` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<ArrowContainer position="bottom">Foobar</ArrowContainer>)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `distance` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<ArrowContainer distance={30}>Foobar</ArrowContainer>)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with other props', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <ArrowContainer as="article" className="k-u-example-class">
            Foobar
          </ArrowContainer>,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
