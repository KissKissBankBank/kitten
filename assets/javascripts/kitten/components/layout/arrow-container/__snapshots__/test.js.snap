// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrowContainer /> with \`centered\` prop matches with snapshot 1`] = `
.c0 {
  position: relative;
  padding: 1.875rem 1.25rem;
  background-color: #f6f6f6;
}

.c1 {
  position: absolute;
  display: block;
  width: 0;
  height: 0;
  border: 0.625rem solid transparent;
  top: calc(50% - 0.625rem);
  left: -1.25rem;
  border-right-color: #f6f6f6;
}

<div
  className="c0"
  color="#f6f6f6"
>
  Foobar
  <span
    className="c1"
    color="#f6f6f6"
    size={10}
  />
</div>
`;

exports[`<ArrowContainer /> with \`color\` prop matches with snapshot 1`] = `
.c0 {
  position: relative;
  padding: 1.875rem 1.25rem;
  background-color: #ff0000;
}

.c1 {
  position: absolute;
  display: block;
  width: 0;
  height: 0;
  border: 0.625rem solid transparent;
  top: 1.25rem;
  left: -1.25rem;
  border-right-color: #ff0000;
}

<div
  className="c0"
  color="#ff0000"
>
  Foobar
  <span
    className="c1"
    color="#ff0000"
    size={10}
  />
</div>
`;

exports[`<ArrowContainer /> with \`distance\` prop matches with snapshot 1`] = `
.c0 {
  position: relative;
  padding: 1.875rem 1.25rem;
  background-color: #f6f6f6;
}

.c1 {
  position: absolute;
  display: block;
  width: 0;
  height: 0;
  border: 0.625rem solid transparent;
  top: 1.875rem;
  left: -1.25rem;
  border-right-color: #f6f6f6;
}

<div
  className="c0"
  color="#f6f6f6"
>
  Foobar
  <span
    className="c1"
    color="#f6f6f6"
    size={10}
  />
</div>
`;

exports[`<ArrowContainer /> with \`position\` prop matches with snapshot 1`] = `
.c0 {
  position: relative;
  padding: 1.875rem 1.25rem;
  background-color: #f6f6f6;
}

.c1 {
  position: absolute;
  display: block;
  width: 0;
  height: 0;
  border: 0.625rem solid transparent;
  bottom: -1.25rem;
  left: 1.25rem;
  border-top-color: #f6f6f6;
}

<div
  className="c0"
  color="#f6f6f6"
>
  Foobar
  <span
    className="c1"
    color="#f6f6f6"
    size={10}
  />
</div>
`;

exports[`<ArrowContainer /> with \`size\` prop matches with snapshot 1`] = `
.c0 {
  position: relative;
  padding: 1.875rem 1.25rem;
  background-color: #f6f6f6;
}

.c1 {
  position: absolute;
  display: block;
  width: 0;
  height: 0;
  border: 1.25rem solid transparent;
  top: 1.25rem;
  left: -2.5rem;
  border-right-color: #f6f6f6;
}

<div
  className="c0"
  color="#f6f6f6"
>
  Foobar
  <span
    className="c1"
    color="#f6f6f6"
    size={20}
  />
</div>
`;

exports[`<ArrowContainer /> with default props matches with snapshot 1`] = `
.c0 {
  position: relative;
  padding: 1.875rem 1.25rem;
  background-color: #f6f6f6;
}

.c1 {
  position: absolute;
  display: block;
  width: 0;
  height: 0;
  border: 0.625rem solid transparent;
  top: 1.25rem;
  left: -1.25rem;
  border-right-color: #f6f6f6;
}

<div
  className="c0"
  color="#f6f6f6"
>
  Foobar
  <span
    className="c1"
    color="#f6f6f6"
    size={10}
  />
</div>
`;

exports[`<ArrowContainer /> with other props matches with snapshot 1`] = `
.c0 {
  position: relative;
  padding: 1.875rem 1.25rem;
  background-color: #f6f6f6;
}

.c1 {
  position: absolute;
  display: block;
  width: 0;
  height: 0;
  border: 0.625rem solid transparent;
  top: 1.25rem;
  left: -1.25rem;
  border-right-color: #f6f6f6;
}

<article
  className="k-u-example-class c0"
  color="#f6f6f6"
>
  Foobar
  <span
    className="c1"
    color="#f6f6f6"
    size={10}
  />
</article>
`;
